country-dashboard/src/pages/Home.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect, useState } from "react";
+import currency from "../data/currency-codes.json";
+import language from "../data/language-codes.json";
+import region from "../data/regions.json";
+import SearchBar from "../components/searchBar";
+import SelectOptions from "../components/selectOptions";
+import { DarkModeSwitch } from "react-toggle-dark-mode";
+import "../style/Home.css";
+import { Data } from "../dataTypes/index";
+import loadingPng from "../Loading State.png";
+
+const Home = () => {
+  // State for dark mode
+  const [darkMode, setDarkMode] = useState<boolean>(false);
+  // State for loading indicator
+  const [loader, setLoader] = useState<boolean>(true);
+
+  // State for search input
+  const [searchValue, setSearchValue] = useState<string>("");
+  // State for search results
+  const [searchData, setSearchData] = useState<Data[]>([]);
+  // State for all countries
+  const [allCountries, setAllCountries] = useState<Data[]>([]);
+  // State for selected currency
+  const [selectedCurrency, setSelectedCurrency] =
+    useState<string>("Any Currency");
+  // State for selected language
+  const [selectedLanguage, setSelectedLanguage] =
+    useState<string>("Any Language");
+  // State for selected region
+  const [selectedRegion, setSelectedRegion] = useState<string>("Any Region");
+
+  // Options for dropdowns
+  const currencyOptions = Object.keys(currency);
+  const languageOptions = Object.keys(language);
+  const baseUrl = "https://restcountries.com/v3.1/";
+
+  // Fetch data on component mount and when dependencies change
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        let url = `${baseUrl}/all`;
+        const response = await fetch(url);
+        const data = await response.json();
+        setAllCountries(data);
+        setLoader(false);
+
+        // Conditionally fetch data based on search or filter criteria
+        if (searchValue !== "") {
```

Comment:
Each of these filter fetches should be in their own useEffect, and then the results should be merged. Otherwise, you end up re-fetching a lot of data you do not need, making the website slow. At the very least, each fetch should be in its own function (i.e `fetchBySearch`). -5